larva/src/main/java/nl/nn/adapterframework/testtool/TestTool.java

```diff
@@ -4203,138 +4205,123 @@ public static String replaceRegularExpression(String string, String from, String
-										errorMessage(
-												"Could not read file '" + partFile
-														+ "': " + e.getMessage(),
-												e, writers);
+										errorMessage("Could not read file '" + partFile+ "': " + e.getMessage(), e, writers);
 									}
 								} else {
 									String string = readFile(partFile, writers);
-									StringPart stringPart = new StringPart(
-											(partName == null ? "string" + j
-													: partName),
-											string);
+									StringPart stringPart = new StringPart((partName == null ? "string" + j : partName), string);
 									parts.add(stringPart);
 								}
 								j++;
 							}
 						}
-						Part allParts[] = new Part[parts.size()];
+						Part[] allParts = new Part[parts.size()];
 						allParts = parts.toArray(allParts);
-						MultipartRequestEntity multipartRequestEntity = new MultipartRequestEntity(
-								allParts, new PostMethod().getParams());
+						MultipartRequestEntity multipartRequestEntity = new MultipartRequestEntity(allParts, new PostMethod().getParams());
 						ByteArrayOutputStream requestContent = new ByteArrayOutputStream();
 						try {
 							multipartRequestEntity.writeRequest(requestContent);
 						} catch (IOException e) {
-							errorMessage(
-									"Could not create multipart: " + e.getMessage(),
-									e, writers);
+							errorMessage("Could not create multipart: " + e.getMessage(), e, writers);
 						}
 						request.setContent(requestContent.toByteArray());
-						request.setContentType(
-								multipartRequestEntity.getContentType());
+						request.setContentType(multipartRequestEntity.getContentType());
 						value = request;
 					}
 					else{
-						MockHttpServletRequest request = new MockHttpServletRequest();
-						value = request;
+						value = new MockHttpServletRequest();
 					}
 				} else {
-					value = properties.getProperty(property + ".param" + i + ".value");
+					value = properties.getProperty(property + _param + i + ".value");
 					if (value == null) {
-						String filename = properties.getProperty(property + ".param" + i + ".valuefile.absolutepath");
+						String filename = properties.getProperty(property + _param + i + ".valuefile.absolutepath");
 						if (filename != null) {
-							value = readFile(filename, writers);
+							value = new FileMessage(new File(filename));
 						} else {
-							String inputStreamFilename = properties.getProperty(property + ".param" + i + ".valuefileinputstream.absolutepath");
+							String inputStreamFilename = properties.getProperty(property + _param + i + ".valuefileinputstream.absolutepath");
 							if (inputStreamFilename != null) {
-								try {
-									value = new FileInputStream(inputStreamFilename);
-								} catch(FileNotFoundException e) {
-									errorMessage("Could not read file '" + inputStreamFilename + "': " + e.getMessage(), e, writers);
-								}
+								errorMessage("valuefileinputstream is no longer supported use valuefile instead", writers);
```

Comment:
I hate to say it, but perhaps we could introduce a 'warningMessage' which is orange (ish).
I'm ok with a TODO here for now.

Reply:
We could still retain the valuefileinputstream for case where one really wants a non-repeatable stream

Reply:
In this case, valuefileinputsteam usage is removed. therefore an errormessage fits better here